### PR TITLE
Add --word-scores to marian-scorer

### PR DESCRIPTION
**Pull request by snukky**
### Description
This PR adds the `--word-scores` option to marian-scorer for printing word-level scores.

Examples:

	./marian-server -c score.yml -t file.in file.out --word-scores
    -1.234479 ||| WordScores= -0.02151 -0.04798 -0.12068 -0.66128 -0.20932 -0.01034 -0.16291 -0.00046
	
	./marian-server -c score.yml -t file.in file.nbest --word-scores --n-best
	0 ||| und das war bei schlechtem Wetter . ||| WordScores= -0.02151 -0.04798 -0.12068 -0.66128 -0.20932 -0.01034 -0.16291 -0.00046 ||| F0= -1.23448 Score= -1.23448 ||| -1.23448

List of changes:
- Added `--word-scores` to marian-scorer.
- For this, I added a new RelationalLoss class that behaves like RescorerLoss, but it does not reduce the loss over time axis; counts are still reduced, to obtain proper sentence lengths.
- If used with `--summarize`, a warning will be printed that word-level scores are not displayed.
- If used with `--normalize`, only sentence scores are normalized. This is consistent with the current behavior of `marian-decoder --word-scores`

Added dependencies: none

### How to test
See added regression tests: https://github.com/marian-nmt/marian-regression-tests/pull/52

### Checklist

- [x] I have tested the code manually
- [x] I have run regression tests
- [x] I have read and followed CONTRIBUTING.md
- [x] I have updated CHANGELOG.md

### CODE REVIEW

File: src/layers/loss.h [hunk 1]
```diff
@@ -437,14 +437,46 @@ class SequenceUnlikelihoodLoss : public CrossEntropyLoss {
...
+  RescorerUnreducedLoss() : CrossEntropyLoss(/*axes=*/{}, /*smoothing=*/0.f, /*factorWeight=*/1.0f) {}
+
+  virtual RationalLoss apply(Logits logits, const Words& labels,
+                             Expr mask = nullptr, Expr labelWeights = nullptr) override {
+    auto loss = CrossEntropyLoss::compute(logits, labels, mask, labelWeights);
+    ABORT_IF(!mask, "Word-level CE from rescorer must have mask");
+    return reduceLabels(loss, mask);  // reduce labels to get sentence lengths
+  }
+
+  RationalLoss reduceLabels(Expr loss, Expr labels) {
```
Comment (emjotde) [inline src/layers/loss.h:468]: This is just a utlity function, right? Not meant for inheriting etc. So I guess that should either be private or actually just merged with `apply()`. I don't see a good reason for having that split into two functions?

Comment (emjotde) [reply src/layers/loss.h:468]: And since we are already talking about merging. Would it make sense to just merge the two classes `RescorerLoss` and `RescorerUnreducedLoss` and just pass a boolean argument `bool wordLevelScores` in order to tell what is being returned and what behavior is expected?

Comment (snukky) [reply src/layers/loss.h:468]: You are totally right, this will be much cleaner. Thanks!

Comment (snukky) [reply src/layers/loss.h]: Done

**Comment by snukky** [comment]
Code updated to have one RescorerLoss class.
